pgrx/src/datum/from.rs

```diff
@@ -412,7 +412,22 @@ impl FromDatum for String {
         is_null: bool,
         typoid: pg_sys::Oid,
     ) -> Option<String> {
-        FromDatum::from_polymorphic_datum(datum, is_null, typoid).map(|s: &str| s.to_owned())
+        if varlena::varatt_is_1b_e(datum.cast_mut_ptr::<pg_sys::varlena>())
+            || varlena::varatt_is_4b_c(datum.cast_mut_ptr::<pg_sys::varlena>())
```

Comment:
I believe these two shouldn't need to be turbofished, as the inference should be effectively infallible due to [function arguments being coercion sites](https://doc.rust-lang.org/reference/type-coercions.html#coercion-sites), which you'll notice is what happens on line 418. It's fine if they are, though.